scripts/simplify_network.py

```diff
@@ -522,6 +523,64 @@ def consense(x):
     return clustering.network, clustering.busmap
 
 
+def merge_isolated_buses(n, drop_p_threshold=0):
+    generators_sum_origin = n.generators.p_nom.sum()
+    load_sum_origin = n.loads_t.p_set.sum().sum()
+
+    # duplicated sub-networks mean that there is at least one interconnection between buses
+    i_islands = n.buses[
+        (~n.buses.duplicated(subset=["sub_network"], keep=False))
+        & (n.buses.carrier == "AC")
+    ].index
+
+    load_sum_isol_origin = n.loads_t.p_set[i_islands].sum().sum()
+
+    # disregard buses having very small attached load when rearranging
+    loads_t_avr = n.loads_t.p_set.mean(axis=0)
```

Comment:
better to make the filter by yearly demand. we can calculate the mean and multiply by 8760 (better than sum as it is reliant to variable length of the dataset)

Reply:
Agree. Fixed

Reply:
ok

> Hello Davide!
> 
> Thank you so much for checking the PR. I'm a bit puzzled by a way CI fails and your comments are very helpful to see a possible workaround for it :) Agree that it absolutely makes sense to decompose the problem

I completely understand.
My feeling is that this issues are the consequence of manually removing data from the dataframes to handle the merge of isolated buses. the clustering procedure does that automatically.

The first option, implying dropping the buses should involve less code and I'd recommend to implement a specific function for that and (hopefully) these issues may disappear

Reply:
> ok
> 
> > Hello Davide!
> > Thank you so much for checking the PR. I'm a bit puzzled by a way CI fails and your comments are very helpful to see a possible workaround for it :) Agree that it absolutely makes sense to decompose the problem
> 
> I completely understand. My feeling is that this issues are the consequence of manually removing data from the dataframes to handle the merge of isolated buses. the clustering procedure does that automatically.
> 
> The first option, implying dropping the buses should involve less code and I'd recommend to implement a specific function for that and (hopefully) these issues may disappear

Have tried to implement the drop option and all the issues were even resolved locally for the tutorial case, but re-appeared after some additional changes. Which confirms that the decomposing approach is a very good idea ;) 

It'll be definitely simpler to debug the dropping part first as compared to the whole approach. Thanks a lot for you support!